### PR TITLE
run0: implement CLI

### PR DESCRIPTION
**Pull request by thomasgouveia**
@sameo , here a basic CLI implementation, which provides the ability to run a container with the following : 

`run0 run -b /path/to/bundle`

I tried to design it so we can easily add more commands in the future, I'm open to any suggestions about my implementation !

### CODE REVIEW

File: run0/src/cli/mod.rs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    pub(crate) command: Commands,
+}
+
+impl Cli {
+    /// Get the command used by the user.
+    ///
+    /// For example, if the user executes the command `run`,
+    /// we dynamically return the command so the `main` can
+    /// execute it.
+    pub fn get_command(self) -> Box<dyn Handler> {
```
Comment (sameo) [inline run0/src/cli/mod.rs]: Please rename this getter to `command`. See https://github.com/rust-lang/rfcs/blob/master/text/0344-conventions-galore.md#gettersetter-apis

File: run0/src/container/mod.rs [hunk 2]
```diff
@@ -14,19 +15,6 @@ mod namespaces;
...
-#[derive(Debug)]
-pub enum Error {
-    OCISpecificationLoad(oci_spec::OciSpecError),
-    OCIInvalidNamespace(oci_spec::runtime::LinuxNamespaceType),
-    ContainerSpawnCommand(unshare::Error),
-    ContainerWaitCommand(std::io::Error),
-    ContainerExit(i32),
-    Unmount(std::io::Error),
-}
-
```
Comment (sameo) [inline run0/src/container/mod.rs:27]: I'd prefer if we keep those errors in the container module, as they're container related.

The cli module would define its own errors as well, that would typically take a container error as an input:

```rust
/// CLI errors
#[derive(Debug)]
pub enum Error {
    ContainerRunError(crate::container::Error),
}
```

File: run0/src/cli/mod.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+/// Each of our commands should be listed in this enumeration with the following format :
+/// CommandName(CommandHandler)
+///
+/// Example:
+///
+/// You want to add the `list` command:
+///
+/// List(ListCommand)
+#[derive(Subcommand, Debug)]
+pub enum Commands {
```
Comment (sameo) [inline run0/src/cli/mod.rs]: `enum Command` please.

**Review by sameo — COMMENTED** [review]
Thanks for the PR @thomasgouveia.

Except for a couple of nits and the change in error handling, it mostly looks good to me.

**Comment by thomasgouveia** [comment]
Hi @sameo, the PR was fixed with your reviews. It should be ready to be merged if my changes are ok for you

**Review by sameo — APPROVED** [review]

**Pull request merged**